### PR TITLE
OAuth input

### PR DESCRIPTION
**Pull request by luhan2017**
Enable OAuthInput in Adaptive

#2333

### CODE REVIEW

**Comment by carlosscastro** [comment]
Would be great to think of a way of testing this by mocking the token retrieval code, but looks good overall!

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/OAuthInput.cs [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+        /// </summary>
+        /// <value>Any additional text to include in the sign-in card.</value>
+        public string Text { get; set; }
+
+        /// <summary>
+        /// Gets or sets the number of milliseconds the prompt waits for the user to authenticate.
+        /// Default is 900,000 (15 minutes).
+        /// </summary>
+        /// <value>The number of milliseconds the prompt waits for the user to authenticate.</value>
+        public double Timeout { get; set; } = 900000;
```
[the remark this answers is not in the document]
Comment (luhan2017) [reply libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/OAuthInput.cs]: My concern for the TimeSpan is that it is not quite straight-forward for declarative. in the OAuthPromptSettings it is set to int, let me set it to int, is that OK for you?

[the remark this answers is not in the document]
Comment (carlosscastro) [reply libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/OAuthInput.cs]: Oh this is a declaratively exposed property. Then yes, as-is is fine, thanks Lu

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/OAuthInput.cs [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+                        opt.RetryPrompt.InputHint = InputHints.AcceptingInput;
+                    }
+                }
+            }
+
+            // Initialize state
+            var timeout = Timeout;
+            var state = dc.ActiveDialog.State;
+            state[PersistedOptions] = opt;
+            state[PersistedState] = new Dictionary<string, object>
```
[the remark this answers is not in the document]
Comment (luhan2017) [reply libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/OAuthInput.cs]: The state is got from dc.ActiveDialog.State, so it should belongs to the specific dialog, I think it is fine.

**Pull request merged**